micall/core/contig_stitcher.py

```diff
@@ -0,0 +1,631 @@
+    """
+
+    for other in aligned_contigs:
+        if overlap(self, other):
+            yield other
+
+
+def find_overlapping_contig(self: AlignedContig, aligned_contigs):
+    """
+    Find the single contig in a collection that overlaps the most with a given contig.
+    It returns the contig with the maximum overlapped reference length with the given contig (self).
+    """
+
+    every = find_all_overlapping_contigs(self, aligned_contigs)
+    return max(every, key=lambda other: other.alignment.ref_length if other else 0, default=None)
+
+
+def calculate_concordance(left: str, right: str) -> List[Fraction]:
+    """
+    Calculate concordance for two given sequences using a sliding average.
+
+    The function compares the two strings character by character, simultaneously from
+    both left to right and right to left, calculating a score that represents a moving
+    average of matches at each position. If characters match at a given position,
+    a score of 1 is added; otherwise, a score of 0 is added. The score is then
+    averaged with the previous scores using a weighted sliding average where the
+    current score has a weight of 1/3 and the accumulated score has a weight of 2/3.
+    This sliding average score is halved and then processed again, but in reverse direction.
+
+    :param left: string representing first sequence
+    :param right: string representing second sequence
+    :return: list representing concordance ratio for each position
+    """
+
+    if len(left) != len(right):
+        raise ValueError("Can only calculate concordance for same sized sequences")
+
+    result: List[Fraction] = [Fraction(0)] * len(left)
+
+    def slide(start, end):
+        scores_sum = Fraction(0)
+        inputs = list(zip(left, right))
+        increment = 1 if start <= end else -1
+
+        for i in range(start, end, increment):
+            (a, b) = inputs[i]
+            current = Fraction(1) if a == b else Fraction(0)
+            scores_sum = (scores_sum * 2 / 3 + current * 1 / 3)
+            result[i] += scores_sum / 2
+
+    # Slide forward, then in reverse, adding the scores at each position.
+    slide(0, len(left))
+    slide(len(left) - 1, -1)
+
+    return result
+
+
+def disambiguate_concordance(concordance: List[float]) -> Iterable[Tuple[float, int]]:
+    for i, x in enumerate(concordance):
+        global_rank = i if i < len(concordance) / 2 else len(concordance) - i - 1
+        yield x, global_rank
+
+
+def concordance_to_cut_points(left_overlap, right_overlap, aligned_left, aligned_right, concordance):
+    """ Determine optimal cut points for stitching based on sequence concordance in the overlap region. """
+
+    concordance_d = list(disambiguate_concordance(concordance))
+    sorted_concordance_indexes = sorted(range(len(concordance)), key=lambda i: concordance_d[i])
+    def remove_dashes(s): return ''.join(c for c in s if c != '-')
```

Comment:
I haven't measured it, but I expect that `replace()` would be faster and easier to understand.
```suggestion
    def remove_dashes(s): return s.replace('-', '')
```

It could probably even be inlined.